lua/cfc_time/server/time/sv_mysql.lua

```diff
@@ -0,0 +1,265 @@
+    query:clearParameters()
+
+    for k, v in pairs( { ... } ) do
+        if isnumber( v ) then
+            query:setNumber( k, v )
+        elseif isstring( v ) then
+            query:setString( k, v )
+        elseif isbool( v ) then
+            query:setBoolean( k, v )
+        elseif v == nil then
+            query:setNull( k )
+        else
+            error( "Wrong data type passed to Prepare statement!: " .. v )
+        end
+    end
+
+    if onSuccess then query.onSuccess = onSuccess end
+
+    return query
+end
+
+function storage.database:onConnected()
+    logger:info( "DB successfully connected! Beginning init..." )
+
+    local transaction = storage:InitTransaction()
+
+    transaction:addQuery( storage:CreateUsersQuery() )
+    transaction:addQuery( storage:CreateSessionsQuery() )
+    transaction:addQuery( storage:SessionCleanupQuery() )
+
+    transaction.onSuccess = function()
+        storage:PrepareStatements()
+    end
+
+    transaction:start()
+end
+
+function storage.database:onConnectionFailed( err )
+    logger:error( "Failed to connect to database!" )
+    logger:fatal( err )
+end
+
+hook.Add( "PostGamemodeLoaded", "CFC_Time_DBInit", function()
+    logger:info( "Gamemoded loaded, beginning database init..." )
+    storage.database:connect()
+end )
+
+-- TODO: Find a better/safer way to do this
+function storage:BuildSessionUpdate( data, id )
+    local updateSection = "UPDATE sessions "
+    local setSection = "SET "
+    local whereSection = string.format(
+        "WHERE id = %s AND realm = '%s'",
+        id, self.realm
+    )
+
+    -- TODO: Have a safeguard here for invalid keys?
+    local count = table.Count( data )
+    local idx = 1
+
+    -- FIXME: This will break because we pass strings/ints interchangeably, but strings need quotes around them
+    for k, v in pairs( data ) do
+        local newSet = k .. " = " .. v
+        
+        if idx ~= count then
+            -- Add a comma if it isn't the last one
+            newSet = newSet .. ","
+        else
+            -- Add a space if it's the last one
+            newSet = newSet .. " "
+        end
+        
+        setSection = setSection .. newSet
+        idx = idx + 1
+    end
+
+    local query = updateSection .. setSection .. whereSection
+
+    logger:info( "Created sessions update query!" )
+    logger:info( query )
+
+    return self.database:escape( query )
+end
+
+--[ API Begins Here ]--
+
+function storage:UpdateBatch( batchData )
+    if not batchData then return end
+    if table.Count( batchData ) == 0 then return end
+
+    local transaction = storage:InitTransaction()
+
+    for sessionId, data in pairs( batchData ) do
+        local updateStr = self:BuildSessionUpdate( data, sessionId )
+        local query = self.database:query( updateStr )
+
+        transaction:addQuery( query )
+    end
+
+    transaction:start()
+end
+
+function storage:GetTotalTime( steamId, callback )
+    local onSuccess = function( _, data )
```

Comment:
```suggestion
    local function onSuccess( _, data )
```

Not sure if this is better, but less characters ;)